Sources/Automerge/Document.swift

```diff
@@ -768,6 +768,61 @@ public final class Document: @unchecked Sendable {
         }
     }
 
+    /// Get the list of marks for a text object at ``Cursor``.
+    ///
+    /// Use:
+    /// ```
+    /// let doc = Document()
+    /// let textId = try doc.putObject(obj: ObjId.ROOT, key: "text", ty: .Text)
+    ///
+    /// let cursor = try doc.cursor(obj: textId, position: 0)
+    /// let marks = try doc.marksAt(obj: textId, cursor: cursor)
+    /// ```
+    ///
+    /// - Parameters:
+    ///   - obj: The identifier of the text object.
+    ///   - cursor: The cursor that indicates a position within the text.
+    ///   - heads: The set of ``ChangeHash`` that represents a point of time in the history the document.
+    /// - Returns: A list of ``Mark`` for the text object at the specified ``Cursor``.
+    public func marksAt(obj: ObjId, cursor: Cursor, heads: Set<ChangeHash>? = nil) throws -> [Mark] {
+        try sync {
+            try self.doc.wrapErrors {
+                try $0.marksCursor(
+                    obj: obj.bytes,
+                    cursor: cursor.bytes,
+                    heads: heads?.map(\.bytes) ?? $0.heads()
+                ).map(Mark.fromFfi)
+            }
+        }
+    }
+
+    /// Get the list of marks for a text object at specified position.
+    ///
+    /// Use:
+    /// ```
+    /// let doc = Document()
+    /// let textId = try doc.putObject(obj: ObjId.ROOT, key: "text", ty: .Text)
+    ///
+    /// let marks = try doc.marksAt(obj: textId, position: 0)
+    /// ```
+    ///
+    /// - Parameters:
+    ///   - obj: The identifier of the text object.
+    ///   - position: The index that indicates the position of the text offset.
+    ///   - heads: The set of ``ChangeHash`` that represents a point of time in the history the document.
```

Comment:
Since this is set up as optional, include some detail about what the default operation does when the value is not provided by the developer (e.g. "uses the latest version of the document")

Reply:
https://github.com/automerge/automerge-swift/pull/186#discussion_r1637157757